### PR TITLE
[Docs] Add/update model monitoring

### PR DESCRIPTION
**Pull request by jillnogold**

### CODE REVIEW

File: docs/monitoring/model-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+```{admonition} Note
+If you are using the CE version, see {ref}`legacy-model-monitoring`.
+```
+## Architecture
+
+<img src="../_static/images/model-monitoring.png" width="1100" >
+
+The model monitoring process flow starts with collecting operational data. The operational data is converted to vectors, which are posted to the Model Server. 
+The model server is then wrapped around a machine learning model that uses a function to calculate predictions based on the available vectors. Next, the model 
+server creates a log for the input and output of the vectors, and the entries are written to the production data stream (a v3io stream). While the model server
```
Comment (gilad-shaham) [inline docs/monitoring/model-monitoring.md]: I don't think v3io stream helps here for clarity

File: docs/monitoring/model-monitoring.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+When you enable model monitoring, you effectively deploy three components:
+- application controller function: handles the monitoring processing and the triggers the apps that trigger the writer. The frequency is determined by `base_period`. 
+- stream function: monitors the log of the data stream. It is triggered when a new log entry is detected. The monitored data is used to create real-time dashboards, detect drift, and analyze performance.
+- writer function: writes to the database and outputs alerts.
+
+## APIs
+
+The model monitoring APIs are configured per project. The APIs are:
+
+- {py:meth}`~mlrun.db.httpdb.HTTPRunDB.enable_model_monitoring` &mdash; brings up the controller and schedules it according to the `base_period`; deploys the writer.
```
Comment (gilad-shaham) [inline docs/monitoring/model-monitoring.md]: Is this in HTTPDB or project?

File: docs/monitoring/model-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- {py:meth}`~mlrun.projects.MlrunProject.disable_model_monitoring` &mdash; disables the controller.
```
Comment (gilad-shaham) [inline docs/monitoring/model-monitoring.md]: let's start with the API to use and at the end how to disable

File: docs/monitoring/model-monitoring.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+- {py:meth}`~mlrun.projects.MlrunProject.disable_model_monitoring` &mdash; disables the controller. 
+- {py:meth}`~mlrun.projects.MlrunProject.set_model_monitoring_function` &mdash; Update or set a monitoring function to the project. (Monitoring does not start until the function is deployed.) 
+- {py:meth}`~mlrun.projects.MlrunProject.create_model_monitoring_function` &mdash; creates a function but does not set it. It's useful for troubleshooting, since it does  not register the function to the project.
+- {py:meth}`~mlrun.projects.MlrunProject.list_model_monitoring_functions` &mdash; Retrieves a list of all the model monitoring functions.
+- {py:meth}`~mlrun.projects.MlrunProject.remove_model_monitoring_function` &mdash; Removes the specified model-monitoring-app function from the project and from the DB.
+- {py:meth}`~mlrun.projects.MlrunProject.set_model_monitoring_credentials` &mdash; Sets the Kafka or SQL credentials to be used by the project's model monitoring infrastructure functions. 
+
+## Configuration flow
+
```
Comment (gilad-shaham) [inline docs/monitoring/model-monitoring.md]: I think code blocks are needed here, not just text

File: docs/monitoring/monitoring-models.ipynb [hunk 5]
```diff
@@ -0,0 +1,317 @@
+{
```
Comment (gilad-shaham) [inline docs/monitoring/monitoring-models.ipynb:1]: This notebook is very hard to follow, I don't see how to get to those screens and what's the context for it. Need to have it as part of the flow
Also, for the dashboards, are these the same for the new infrastructure?

File: docs/monitoring/untitled
Comment (gilad-shaham) [inline docs/monitoring/untitled:1]: What is that file?

File: docs/tutorials/05-model-monitoring.ipynb [hunk 6]
```diff
@@ -2,396 +2,1398 @@
...
    "cell_type": "markdown",
+   "id": "9c8944e0-565a-4bc2-9515-e9a8f9724fb8",
    "metadata": {},
    "source": [
-    "```{admonition} Note\n",
-    "This tutorial does not focus on training a model. Instead, it starts with a trained model and its corresponding training dataset.\n",
-    "```\n",
+    "## Enable model monitoring\n",
+    "\n",
+    "Model monitoring is enabled per project. {py:meth}`~mlrun.db.httpdb.HTTPRunDB.enable_model_monitoring` brings up the controller and schedules it according to the `base_period`, and deploys the writer.\n",
```
Comment (gilad-shaham) [inline docs/tutorials/05-model-monitoring.ipynb]: again, this is referencing httpd and the code goes to project, so need to be clear what to use

File: docs/tutorials/05-model-monitoring.ipynb [hunk 7]
```diff
@@ -2,396 +2,1398 @@
...
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "At this point, you should see the newly deployed model server, as well as a `model-monitoring-stream`, and a scheduled job (in yellow). The `model-monitoring-stream` collects, processes, and saves the incoming requests to the model server. The scheduled job does the actual calculation (by default every hour).\n",
+    "# register the first app named \"demo_app\"\n",
+    "my_app = project.set_model_monitoring_function(\n",
+    "    func=\"./assets/demo_app.py\",\n",
```
Comment (gilad-shaham) [inline docs/tutorials/05-model-monitoring.ipynb]: where is this python file coming from?

File: docs/tutorials/05-model-monitoring.ipynb [hunk 8]
```diff
@@ -2,396 +2,1398 @@
...
   {
    "cell_type": "markdown",
+   "id": "ea8f6c9f",
    "metadata": {},
    "source": [
     "# Model monitoring and drift detection\n",
     "\n",
-    "This tutorial illustrates leveraging the model monitoring capabilities of MLRun to deploy a model to a live endpoint and calculate data drift.\n",
+    "This tutorial illustrates the basic model monitoring capabilities of MLRun: deploying a model to a live endpoint and \n",
+    "calculating data drift.\n",
```
Comment (gilad-shaham) [inline docs/tutorials/05-model-monitoring.ipynb:11]: The flow here needs to be better explained, it starts with real-time and then goes to batch, we should better explain to the user what they are expected to see and the general idea of the flow

**Review by gilad-shaham — CHANGES_REQUESTED** [review]

File: docs/monitoring/model-monitoring.md (hunk 2 above)
Comment (jillnogold) [reply docs/monitoring/model-monitoring.md]: now project!

File: docs/monitoring/untitled
Comment (jillnogold) [reply docs/monitoring/untitled:1]: oops... deleted

File: docs/tutorials/05-model-monitoring.ipynb (hunk 6 above)
Comment (jillnogold) [reply docs/tutorials/05-model-monitoring.ipynb]: fixed

**Comment by ghost** [comment]
Hi, I wanted to add that this PR looks great, the documentation is very detailed about all the capabilities of the model monitoring!

However, I think that we miss some examples of applications that can be used in the new model monitoring. Since the new model monitoring is based on user defined applications, it will be helpful to provide some examples that show how the application function should look like (it needs to have fixed parameters and return some fixed values).
I will add 2 examples notebook that I used when I was asked to test it. Feel free to use any part of it or take inspiration from them. Just note that:
- One notebook uses Evidently in the application function to perform the model monitoring.
- The second one also uses Evidently in the application function, but it tested how to monitor LLM models. So the Evidently monitoring is more 'basic' as we just wanted to test if there is a way to monitor 'text' data since it's LLM. The notebook also contains some workaround that is the process of being solved (or maybe it's already solved) - it logs a 'dummy' model to create a model endpoint. This won't be needed in the future.

If you need any more info about the new model monitoring and the applications- I can try to help but @Eyal-Danieli is the address

Examples in here https://mckinsey.box.com/s/r8ovuiu8bywtkdq2kfqpeqln9lfswqzy

**Comment by jillnogold** [comment]
@amit-elbaz Are your comments all relevant to v1.6? I want to get the first updates out soon, and then update for 1.7.
Thanks

**Comment by ghost** [comment]
@jillnogold I think this starts in 1.7. @Eyal-Danieli Can you confirm?

File: docs/monitoring/model-monitoring.md (hunk 4 above)
Comment (jillnogold) [reply docs/monitoring/model-monitoring.md]: added

File: docs/monitoring/monitoring-models.ipynb (hunk 5 above)
Comment (jillnogold) [reply docs/monitoring/monitoring-models.ipynb:1]: @gilad-shaham It's an overview. Specific graphs are mentioned in the user-flow/tutorial. Dashboards are updated. (This page was pre-existing, I just updated.)

File: docs/tutorials/05-model-monitoring.ipynb (hunk 7 above)
Comment (jillnogold) [reply docs/tutorials/05-model-monitoring.ipynb]: added

File: docs/tutorials/05-model-monitoring.ipynb (hunk 8 above)
Comment (jillnogold) [reply docs/tutorials/05-model-monitoring.ipynb:11]: added more explanation

File: docs/tutorials/05-model-monitoring.ipynb [hunk 9]
```diff
@@ -2,452 +2,1478 @@
...
-    "    model_file=model_path,\n",
-    "    framework=\"sklearn\",\n",
-    "    training_set=pd.read_csv(training_set_path),\n",
-    "    label_column=\"label\",\n",
-    ")"
+    "## Import, enable monitoring, and deploy the serving function\n",
+    "\n",
+    "Import the [model server function](https://www.mlrun.org/hub/functions/master/v2-model-server/latest/example/)  from the MLRun Function Hub, add the model that was logged via experiment tracking, and enable drift detection.\n",
+    "\n",
+    "The core line here is `serving_fn.set_tracking()`, which creates the required infrastructure behind the scenes to perform drift detection. \n",
```
Comment (Eyal-Danieli) [inline docs/tutorials/05-model-monitoring.ipynb]: it's not accurate. the `enable_model_monitoring()` API creates the infra behind the scenes. Applying `set_tracking()` on the serving function is indeed required but the meaning of this operation is that the relevant models will be tracked, and eventually we will be able to perform drift detection on these models.

**Review by Eyal-Danieli — COMMENTED** [review]
a single comment

File: docs/tutorials/05-model-monitoring.ipynb (hunk 9 above)
Comment (jillnogold) [reply docs/tutorials/05-model-monitoring.ipynb]: updated

**Review by Eyal-Danieli — APPROVED** [review]
